djstripe/models/billing.py

```diff
@@ -985,22 +1006,6 @@ def _attach_objects_post_save_hook(self, cls, data, pending_relations=None):
                 cls._stripe_object_to_tax_rates(target_cls=TaxRate, data=data)
             )
 
-    @classmethod
-    def sync_from_stripe_data(cls, data):
-        invoice_data = data.get("invoice")
-
-        if invoice_data:
-            # sync the Invoice first if it doesn't yet exist in our DB
-            # to avoid recursive Charge/Invoice loop
-            invoice_id = cls._id_from_data(invoice_data)
-            if not Invoice.objects.filter(id=invoice_id).exists():
-                if invoice_id == invoice_data:
-                    # we only have the id, fetch the full data
-                    invoice_data = Invoice(id=invoice_id).api_retrieve()
-                Invoice.sync_from_stripe_data(data=invoice_data)
-
-        return super().sync_from_stripe_data(data)
-
```

Comment:
Can you explain why it's safe to remove this now?

Reply:
The reason why the generic scenario of 2 models having `nullable FKs` to each other will not result in infinite recursive sync is because of:

1.  `FKs` are [now](https://github.com/dj-stripe/dj-stripe/pull/1381/files#diff-92652cc100faa74b696687ae45deb269ec9937e9c1a89d77b69e489c62cf67d5R433-R439) getting added to the `current_ids` set when they are either getting fetched or created (by `_get_or_create_from_stripe_object`) and then removed from `current_ids`once that is done. 
2. This ensures that 2 or more models that need the same `RelatedModel` to exist will not keep on requesting it or each other, as the case maybe, (and hence avoid the infinite loop) and continue processing their other model fields. This happens because on the next loop of `_get_or_create_from_stripe_object` the function [skips processing](https://github.com/dj-stripe/dj-stripe/blob/master/djstripe/models/base.py#L423) that field if it is in the `current_ids` set (which it already is in as we do that just before calling `_get_or_create_from_stripe_object`). Hence, all the models will simply continue processing their other model fields and save the `FK` as null for the time being. 
3. Once the model is saved, its `_attach_objects_post_save_hook` will get fired which would then kick-off processing all ids in `current_ids` which should exist by then (because their respective models would have either been saved already or [this method](https://github.com/dj-stripe/dj-stripe/blob/master/djstripe/models/base.py#L468) will wait till that happens). The `FK` will then get updated as expected!

This is how I get around the infinite loop by basically delaying saving all `nullable FKs`. The only requirement is to have them be `nullable` and this method should work for all, at least in my opinion.

Regarding tests, for `Transfer` and `TransferReversal` they should be complete. They are incomplete for `ApplicationFee` and `ApplicationFeeReversal` right now. I can comment out the custom methods I added until tests can be added for them.

I haven't started working on using the API to test and replace mocks so far.

Reply:
Hey guys, we have a large number of webhook events failing because of this infinite loop, and are looking forward to this change. Is this expected to be released in 2.6.0 or could it come sooner? Thanks a lot, and great work identifying the source of the issue. :slightly_smiling_face:

Reply:
@ryanenian I will try and add the  changes as suggested by Jerome by this weekend. Let's hope everything works fine after this PR gets merged.

Reply:
much appreciated @arnav13081994!